trajopt/src/collision_terms.cpp

```diff
@@ -1398,16 +1440,31 @@ void CastCollisionEvaluator::CalcCollisions(const Eigen::Ref<const Eigen::Vector
       return;
     }
 
+    // Contains the contact distance threshold and coefficient for the given link pair
+    const double dist = collision_config_->contact_manager_config.margin_data.getPairCollisionMargin(pair.first.first,
+                                                                                                     pair.first.second);
+    const double coeff =
+        collision_config_->collision_coeff_data.getPairCollisionCoeff(pair.first.first, pair.first.second);
+    const Eigen::Vector3d data = { dist, collision_config_->collision_margin_buffer, coeff };
+
     // Don't include contacts at the fixed state
-    const std::array<double, 2>& data =
-        getSafetyMarginData()->getPairSafetyMarginData(pair.first.first, pair.first.second);
-    removeInvalidContactResults(pair.second, data);
+#ifndef NDEBUG
+    tesseract_collision::ContactResultVector copy{ pair.second };
+#endif
+
+    trajopt_common::removeInvalidContactResults(pair.second, data, position_vars_fixed_);
+
+#ifndef NDEBUG
+    std::array<double, 2> temp_data{ dist, coeff };
+    removeInvalidContactResults(copy, temp_data);
+    assert(copy.size() == pair.second.size());
+#endif
```

Comment:
Same comments as above about revising the packaging of `data` in a vector, the purpose of the debug section, etc.